src/main/java/org/folio/search/integration/KafkaMessageListener.java

```diff
@@ -39,7 +44,19 @@ public void handleEvents(List<ResourceEventBody> events) {
       .map(KafkaMessageListener::asInstanceResource)
       .collect(toList());
 
-    indexService.indexResources(resources);
+    resources.stream()
+      .collect(groupingBy(ResourceEventBody::getTenant))
+      .forEach((tenant, eventsForTenant) -> {
+        try {
+          log.info("Processing [{}] events for tenant [{}]", eventsForTenant.size(), tenant);
+          // This needed to inject FOLIO specific values to properly work with DB
+          beginFolioExecutionContext(new AsyncFolioExecutionContext(tenant, moduleMetadata));
```

Comment:
I guess this part should be tested in unit tests

Reply:
good point, I've added a unit test for it.